### DISCUSSION TITLE
Update record in database using sqlalchemy

### DISCUSSION THREAD

**Discussion by nabarunn** [sqlalchemy/sqlalchemy#9800] [Usage Questions]
I have retrieved a record from database, converted it into a pydantic object and have moved it around different classes updating some of the record's fields. Finally I converted it back to SqlAlchemy model want to update it in the database. How can I achieve it? Simple solutions like in the following will not work for me.

1.  db.query(Model).filter(Model.field1 == 2).first().update(update_dict)
2.  model =  db.query(Model).filter(Model.field1 == 2).first()
     model.field1 = 2
     db.commit()
     
Adding:- 

Database - Oracle Sql
SqlAlchemy version =  2.0.13
OS = MacOs

**Comment by johaven** [comment]
I think we'll tell you that it lacks context :D
Version, Os, Database ? Request logs?

**Reply by nabarunn** [reply]
Sure I will edit with version and os

**Reply by zzzeek** [reply]
that second example should work though, loaded from the ORM , update and commit.  turn on echo=True and see what the SQL output is

**Reply by nabarunn** [reply]
@zzzeek thanks for your suggestion to turn on echo which made me understand how filter and update works together to update a record. And following is what worked for me.

db.query(User).filter(User.id == modefied_user.id).update(values=modified_user.dict(by_alias=False))
db.commit()

In the above snippet - 
User -> model
modified_user -> pydantic object
